### PR TITLE
[New] add `--entrypoints-legacy` CLI command

### LINKED ISSUE #153: confusing resolution failure
In https://github.com/ljharb/eslint-plugin-import/tree/types, cd into `utils/`, run `npm install`, and then run `npx @arethetypeswrong/cli -P`.

It gives resolution failed, but I'm not sure why.

### PR DESCRIPTION
**Pull request by ljharb**
Note that the snapshots here are incorrect - despite these changes working as expected on the actual project/version in question, it's not working when i run the tests locally.

Closes #153.

### CODE REVIEW

File: packages/core/src/checkPackage.ts [hunk 1]
```diff
@@ -47,7 +57,14 @@ export async function checkPackage(pkg: Package, options?: CheckPackageOptions):
...
 
   const hosts = createCompilerHosts(pkg);
-  const entrypointResolutions = getEntrypointInfo(packageName, pkg, hosts, options);
+  let files: string[] | undefined = undefined;
+  if (options?.entrypointsLegacy && dir) {
+    const path = join(process.cwd(), dir);
+    const tree = await new Arborist().loadActual({ path });
+    const extensions = new Set([".js", ".jsx", ".ts", ".tsx", ".mjs", ".cjs", ".mts", ".cts"]);
+    files = (await packlist(tree, { path })).filter(x => !x.endsWith('.d.ts') && extensions.has(extname(x))).map(x => `./${x}`);
+  }
```
Comment (ljharb) [inline packages/core/src/checkPackage.ts:66]: this logic really shouldn't be here - it should be inside `getEntrypointInfo`, but it seemed like making it (and getEntrypoints) an async function was a larger change than i should be making.

happy to do that, tho, if you prefer.

**Review by andrewbranch — COMMENTED** [review]
`@arethetypeswrong/core` always receives or fetches an npm tarball as an input. Isn’t using packlist/arborist redundant with that input? Can we just use all files it can see in the `Package` object?

**Comment by ljharb** [comment]
ah, hmm, i suppose that's true - i was assuming it runs in-situ, but i guess the pack option takes care of that?

I'll refactor to just look at the files, and that should remove the need for `path` as well as the added deps.

**Comment by ljharb** [comment]
thanks, that's much simpler. the snapshot still isn't matching the actual behavior of the built cli, tho.

**Comment by andrewbranch** [comment]
I checked this out locally, ran tests, and got a new snapshot:

````
❯ git diff --staged
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
added: packages/cli/test/snapshots/eslint-module-utils@2.8.1.tgz --entrypoints-legacy --ignore-rules=cjs-only-exports-default.md
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
@ packages/cli/test/snapshots/eslint-module-utils@2.8.1.tgz --entrypoints-legacy --ignore-rules=cjs-only-exports-default.md:4 @
# eslint-module-utils@2.8.1.tgz --entrypoints-legacy --ignore-rules=cjs-only-exports-default

```
$ attw eslint-module-utils@2.8.1.tgz --entrypoints-legacy --ignore-rules=cjs-only-exports-default


eslint-module-utils v2.8.1

Build tools:
- typescript@next

 (ignoring rules: 'cjs-only-exports-default')

 No problems found 🌟


┌─────────────────────────────────────────┬────────┬───────────────────┬───────────────────┬─────────┐
│                                         │ node10 │ node16 (from CJS) │ node16 (from ESM) │ bundler │
├─────────────────────────────────────────┼────────┼───────────────────┼───────────────────┼─────────┤
│ "eslint-module-utils/declaredScope.js"  │ 🟢     │ 🟢 (CJS)          │ 🟢 (CJS)          │ 🟢      │
├─────────────────────────────────────────┼────────┼───────────────────┼───────────────────┼─────────┤
│ "eslint-module-utils/hash.js"           │ 🟢     │ 🟢 (CJS)          │ 🟢 (CJS)          │ 🟢      │
├─────────────────────────────────────────┼────────┼───────────────────┼───────────────────┼─────────┤
│ "eslint-module-utils/ignore.js"         │ 🟢     │ 🟢 (CJS)          │ 🟢 (CJS)          │ 🟢      │
├─────────────────────────────────────────┼────────┼───────────────────┼───────────────────┼─────────┤
│ "eslint-module-utils/module-require.js" │ 🟢     │ 🟢 (CJS)          │ 🟢 (CJS)          │ 🟢      │
├─────────────────────────────────────────┼────────┼───────────────────┼───────────────────┼─────────┤
│ "eslint-module-utils/ModuleCache.js"    │ 🟢     │ 🟢 (CJS)          │ 🟢 (CJS)          │ 🟢      │
├─────────────────────────────────────────┼────────┼───────────────────┼───────────────────┼─────────┤
│ "eslint-module-utils/moduleVisitor.js"  │ 🟢     │ 🟢 (CJS)          │ 🟢 (CJS)          │ 🟢      │
├─────────────────────────────────────────┼────────┼───────────────────┼───────────────────┼─────────┤
│ "eslint-module-utils/parse.js"          │ 🟢     │ 🟢 (CJS)          │ 🟢 (CJS)          │ 🟢      │
├─────────────────────────────────────────┼────────┼───────────────────┼───────────────────┼─────────┤
│ "eslint-module-utils/pkgDir.js"         │ 🟢     │ 🟢 (CJS)          │ 🟢 (CJS)          │ 🟢      │
├─────────────────────────────────────────┼────────┼───────────────────┼───────────────────┼─────────┤
│ "eslint-module-utils/pkgUp.js"          │ 🟢     │ 🟢 (CJS)          │ 🟢 (CJS)          │ 🟢      │
├─────────────────────────────────────────┼────────┼───────────────────┼───────────────────┼─────────┤
│ "eslint-module-utils/readPkgUp.js"      │ 🟢     │ 🟢 (CJS)          │ 🟢 (CJS)          │ 🟢      │
├─────────────────────────────────────────┼────────┼───────────────────┼───────────────────┼─────────┤
│ "eslint-module-utils/resolve.js"        │ 🟢     │ 🟢 (CJS)          │ 🟢 (CJS)          │ 🟢      │
├─────────────────────────────────────────┼────────┼───────────────────┼───────────────────┼─────────┤
│ "eslint-module-utils/unambiguous.js"    │ 🟢     │ 🟢 (CJS)          │ 🟢 (CJS)          │ 🟢      │
├─────────────────────────────────────────┼────────┼───────────────────┼───────────────────┼─────────┤
│ "eslint-module-utils/visit.js"          │ 🟢     │ 🟢 (CJS)          │ 🟢 (CJS)          │ 🟢      │
└─────────────────────────────────────────┴────────┴───────────────────┴───────────────────┴─────────┘


```

Exit code: 0
\ No newline at end of file
````

I think this looks right? What are you getting?

**Comment by ljharb** [comment]
That's totally right! what i get is what's in this PR :-/

**Comment by ljharb** [comment]
I was able to effectively start from scratch and get the md to update, but the json snapshot data still doesn't look right.

**Review by andrewbranch — CHANGES_REQUESTED** [review]
Almost ready to go; if you wouldn’t mind adding something to the CLI README too, that would be great. Thanks!

**Comment by andrewbranch** [comment]
> the json snapshot data still doesn't look right

The JSON snapshot is running the core library with default options; only the CLI test is getting the `entrypointsLegacy` option.

File: packages/core/src/internal/getEntrypointInfo.ts [hunk 2]
```diff
@@ -13,13 +13,21 @@ import type {
...
   if (options?.entrypoints) {
     return options.entrypoints.map((e) => formatEntrypointString(e, fs.packageName));
   }
   if (exportsObject === undefined && fs) {
-    const proxies = getProxyDirectories(`/node_modules/${fs.packageName}`, fs);
+    const rootDir = `/node_modules/${fs.packageName}`;
+    const proxies = getProxyDirectories(rootDir, fs);
     if (proxies.length === 0) {
+      if (options?.entrypointsLegacy) {
+        return fs.listFiles()
```
[the remark this answers is not in the document]
Comment (ljharb) [reply packages/core/src/internal/getEntrypointInfo.ts:27]: tbh I'm not actually sure what "proxy directories" is for, it seems to always be empty in my experience - as for include/exclude, i guess it does make sense to explicitly *exclude* things from the legacy list, but i'm not sure what the point would be for `include` - what would you include that's not already there?

File: packages/core/src/internal/getEntrypointInfo.ts [hunk 3]
```diff
@@ -13,13 +13,21 @@ import type {
...
+          .filter(f => !f.endsWith(".d.ts") && extensions.has(f.slice(f.lastIndexOf("."))))
```
[the remark this answers is not in the document]
Comment (ljharb) [reply packages/core/src/internal/getEntrypointInfo.ts:28]: ah perfect, thanks

File: packages/core/src/internal/getEntrypointInfo.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (andrewbranch) [reply packages/core/src/internal/getEntrypointInfo.ts:27]: Probably nothing, but maybe files that have non-standard/non-JS file extensions

**Comment by ljharb** [comment]
Updated except for https://github.com/arethetypeswrong/arethetypeswrong.github.io/pull/156#discussion_r1504670172

I'm happy to expand the functionality, in this PR, so that `--exclude-entrypoints` *can* be used in concert with `--entrypoints-legacy` - but it should probably be an error to use `--entrypoints` or `--include-entrypoints` with it, since there's nothing that can possibly be included that's not already in the file list?

**Comment by andrewbranch** [comment]
Thanks @ljharb!

**Pull request merged**